src/core/src/op/is_finite.cpp

```diff
@@ -35,3 +38,70 @@ bool ov::op::v10::IsFinite::visit_attributes(AttributeVisitor& visitor) {
     OV_OP_SCOPE(v10_IsFinite_visit_attributes);
     return true;
 }
+
+namespace {
+template <element::Type_t ET>
+bool evaluate_exec(const HostTensorPtr& input, const HostTensorPtr& output) {
+    ngraph::runtime::reference::is_finite(input->get_data_ptr<ET>(),
+                                          output->get_data_ptr<element::Type_t::boolean>(),
+                                          shape_size(input->get_shape()));
+    return true;
+}
+
+#define IS_NAN_TYPE_CASE(a, ...)                                \
+    case element::Type_t::a: {                                  \
+        OV_OP_SCOPE(OV_PP_CAT3(evaluate_exec_is_finite, _, a)); \
+        rc = evaluate_exec<element::Type_t::a>(__VA_ARGS__);    \
+    } break
+
+template <element::Type_t ET>
+bool evaluate(const HostTensorPtr& input, const HostTensorPtr& output) {
+    bool rc = true;
+    switch (input->get_element_type()) {
+        IS_NAN_TYPE_CASE(bf16, input, output);
+        IS_NAN_TYPE_CASE(f16, input, output);
+        IS_NAN_TYPE_CASE(f32, input, output);
+        IS_NAN_TYPE_CASE(f64, input, output);
+    default:
```

Comment:
Do we need all these data types?

Reply:
I think so, based on specification, all 3 OP require all float-like datatypes. For example, IsFinite: https://www.tensorflow.org/api_docs/python/tf/math/is_finite#args